File: Frontend/angular-app/src/app/modules/admin/repositories/person-customers/commands/update-person-customer-command.ts
import {Inject} from "@angular/core";
import {ServiceResult} from "../../../../../core/models/service-result";
import {IRequest, IRequestHandler} from "../../../../../core/services/mediator/interfaces";
import {HttpService} from "../../../../../core/services/http/http.service";
import {ApplicationError} from "../../../../../core/exceptions/application-error";
import {MediatorHandler} from "../../../../../core/services/mediator/decorator";
import {PersonCustomer} from "../../../entities/person-customer";
import {ValidationRule} from "../../../../../core/validation/validation-rule";
import {HttpRequest} from "../../../../../core/services/http/http-request";
import {NotificationService} from "../../../../../shared/services/notification/notification.service";

export class UpdatePersonCustomerCommand extends IRequest<UpdatePersonCustomerCommand, PersonCustomer> {
  public id: number | undefined = undefined;
  public personId!: number;
  public customerTypeBaseId!: number;
  public customerCode!: string;
  public curentExpertId!: number;
  public economicCode!: string;
  public description!: string;

  constructor() {
    super();
  }

  mapFrom(entity: PersonCustomer): UpdatePersonCustomerCommand {
    this.mapBasics(entity,this)
    return this;
  }

  mapTo(): PersonCustomer {
    throw new ApplicationError(UpdatePersonCustomerCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  get url(): string {
    return "/admin/personCustomers/update";
  }

  get validationRules(): ValidationRule[] {
    return [];
  }
}

@MediatorHandler(UpdatePersonCustomerCommandHandler.name)
export class UpdatePersonCustomerCommandHandler implements IRequestHandler<UpdatePersonCustomerCommand, PersonCustomer> {
  constructor(
    @Inject(HttpService) private _httpService: HttpService,
    @Inject(NotificationService) private _notificationService: NotificationService,

  ) {
  }

  async Handle(request: UpdatePersonCustomerCommand): Promise<PersonCustomer> {
    let httpRequest: HttpRequest<UpdatePersonCustomerCommand> = new HttpRequest<UpdatePersonCustomerCommand>(request.url, request);
    // httpRequest.Query  += ;
    // httpRequest.Headers = httpRequest.Headers.append()

    return await this._httpService.Put<UpdatePersonCustomerCommand, ServiceResult<PersonCustomer>>(httpRequest).toPromise().then(response => {
      this._notificationService.showSuccessMessage()
      return response.objResult
    })

  }
}

File: Backend/Microservices/Admin/Eefa.Admin.Application.Refactor/UseCases/PersonCustomer/Commands/Update/UpdatePersonCustomerCommand.cs
using AutoMapper;
using MediatR;
using System.Threading;
using System.Threading.Tasks;

public class UpdatePersonCustomerCommand : IRequest<ServiceResult<PersonCustomerModel>>,
    IMapFrom<Customer>
{
    public int Id { get; set; }
    public int PersonId { get; set; } = default!;
    public int CustomerTypeBaseId { get; set; } = default!;
    public string CustomerCode { get; set; } = default!;
    public int CurentExpertId { get; set; } = default!;
    public string? EconomicCode { get; set; }
    public string? Description { get; set; }


    public void Mapping(Profile profile)
    {
        profile.CreateMap<UpdatePersonCustomerCommand, Customer>()
            .IgnoreAllNonExisting();
    }
}

public class UpdatePersonCustomerCommandHandler : IRequestHandler<UpdatePersonCustomerCommand, ServiceResult<PersonCustomerModel>>
{
    private readonly IUnitOfWork _unitOfWork;
    private readonly IMapper _mapper;

    public UpdatePersonCustomerCommandHandler(IUnitOfWork unitOfWork, IMapper mapper)
    {
        _unitOfWork = unitOfWork;
        _mapper = mapper;
    }

    public async Task<ServiceResult<PersonCustomerModel>> Handle(UpdatePersonCustomerCommand request,
        CancellationToken cancellationToken)
    {
        Customer entity = await _unitOfWork.Customers.GetByIdAsync(request.Id);

        _mapper.Map(request, entity);

        _unitOfWork.Customers.Update(entity);
        await _unitOfWork.SaveChangesAsync(cancellationToken);

        return ServiceResult.Success(_mapper.Map<PersonCustomerModel>(entity));
    }
}